### PR TITLE
Bulk import error handling

### LINKED ISSUE #673: When bulk importing judgments, processing a PDF sometimes fails the whole batch without clarity on which file is failing
* better error reporting on a per-case basis
* when importing from a PDF, better error handling if we can't extract text from the PDF

Errors:

* #675 
* #674 
* #676

### LINKED ISSUE #675: CalledProcessError: Command '['pdftotext', '-enc', 'UTF-8', '/tmp/tmpc9nglzbl', '-']' returned non-zero exit status 1.
Sentry Issue: [LAWLIBRARY-13](https://sentry.io/organizations/lawsafrica/issues/3710116386/?referrer=github_integration)

```
CalledProcessError: Command '['pdftotext', '-enc', 'UTF-8', '/tmp/tmpc9nglzbl', '-']' returned non-zero exit status 1.
(15 additional frame(s) were not displayed)
...
  File "peachjam/models/core_document_model.py", line 276, in extract_citations
    return citation_analyser.extract_citations(self)
  File "peachjam/analysis/citations.py", line 24, in extract_citations
    return self.extract_citations_from_source_file(document)
  File "peachjam/analysis/citations.py", line 42, in extract_citations_from_source_file
    text = pdf_to_text(tmp.name)
  File "docpipe/pdf.py", line 30, in pdf_to_text
    result = subprocess.run(cmd, capture_output=True, check=True)
  File "gevent/subprocess.py", line 2014, in run
    raise _with_stdout_stderr(CalledProcessError(retcode, process.args, stdout), stderr)
```

### LINKED ISSUE #676: CalledProcessError: Command '['soffice', '--headless', '--convert-to', 'html', '--outdir', '/tmp/tmpwwappx50', '/tmp/...
Sentry Issue: [LAWLIBRARY-16](https://sentry.io/organizations/lawsafrica/issues/3727260085/?referrer=github_integration)

```
CalledProcessError: Command '['soffice', '--headless', '--convert-to', 'html', '--outdir', '/tmp/tmpwwappx50', '/tmp/tmpwwappx50/tmpiqg43ufr.docx']' returned non-zero exit status 1.
(16 additional frame(s) were not displayed)
...
  File "docpipe/pipeline.py", line 41, in __call__
    stage(context)
  File "docpipe/soffice.py", line 24, in __call__
    primary, files = soffice_convert(context.source_file, suffix, 'html')
  File "docpipe/soffice.py", line 51, in soffice_convert
    soffice(["--convert-to", outsuffix, "--outdir", tmpdir, f.name])
  File "docpipe/soffice.py", line 68, in soffice
    result = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True,
  File "gevent/subprocess.py", line 2014, in run
    raise _with_stdout_stderr(CalledProcessError(retcode, process.args, stdout), stderr)
```

### PR DESCRIPTION
**Pull request by actlikewill**
- Adds better error handling for bulk imports

closes #673
closes #675
closes #676

### CODE REVIEW

**Comment by actlikewill** [comment]
@longhotsummer The reason we don't see errors in the preview is because the error is triggered in the after_save method after we have done our validations. I have added some code that downloads the file in the SourceFile clean method and calls the pdf_to_text method there. This is working but it feels really inefficient because we will download the file twice, once to validate and the second to save. I want to see if there a way we can  dowload the file once, hold on to the temporary file and reuse it when we do the instance save. I welcome any suggestions :bulb: :thinking:

**Comment by actlikewill** [comment]
@ro6ley Here is the spreadsheet with a corrupt docx file: https://docs.google.com/spreadsheets/d/1KO1tL1BN0V4yu-n4_knQYG4pBXlgomaIv7hxHjR5xGo/edit?usp=sharing

**Comment by actlikewill** [comment]
@ro6ley I found that the issue with the source file conversion was because we were passing a tempfile to the soffice_convert. Using an actual file on disk allows the error to trigger. I have changed the code accordingly.

File: peachjam/resources.py [hunk 1]
```diff
@@ -65,11 +72,27 @@ def clean(self, value, row=None, **kwargs):
...
+                        soffice_convert(file, suffix, "html")
             return source_url
-        except requests.exceptions.RequestException as e:
-            raise ValidationError(e)
+        except (
+            requests.exceptions.RequestException,
+            CalledProcessError,
+            KeyError,
+        ) as e:
+            raise ValidationError(f"source file error {e}")
```
Comment (ro6ley) [inline peachjam/resources.py:95]:
```suggestion
            raise ValidationError(f"Error processing source file: {e}")

```

**Review by ro6ley — DISMISSED** [review]

**Review by ro6ley — DISMISSED** [review]

**Review by ro6ley — APPROVED** [review]

**Pull request merged**